File: src/main/java/hilfling/backend/hilfling/controllers/GangController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.Gang;
import hilfling.backend.hilfling.service.GangService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/gangs")
public class GangController extends GenericBaseControllerImplementation<Gang> {
    @Autowired
    private GangService service;
    public GangService getService() {
        return service;
    }

}

File: src/main/kotlin/hilfling/backend/hilfling/controller/GangController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.*
import hilfling.backend.hilfling.repository.*
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/gangs")
class GangController {
    val repository = GangRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Int): Gang? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll(): List<Gang> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("gang") gang: Gang
    ): Gang {
        return repository.create(
                gang
        )
    }
}